src/mirage_channel.ml

```diff
@@ -71,11 +71,19 @@ module type S = sig
   (** [read_exactly len t] reads [len] bytes from the channel [t] or fails
       with [Eof]. *)
 
-  val read_line: t -> (buffer list Mirage_flow.or_eof, error) result io
-  (** [read_line t] reads a line of input, which is terminated
-      either by a CRLF sequence, or the end of the channel (which
-      counts as a line).  @return Returns a list of views that
-      terminates at EOF. *)
+  val read_line: ?len:int -> t -> (buffer list Mirage_flow.or_eof, error) result io
+  (** [read_line t] reads a line of input which is terminated either by a CRLF
+      sequence, or the end of the channel (which counts as a line).
+
+      If [?len] is provided then the maximum length of the line returned will be
+      [len] bytes. If the line is longer than [len] then an error will be
+      returned.
```

Comment:
Should specify what happens to the input pointer when an error is returned -- is it at the original place, or has it advanced to the limit of the line length?  Is the channel usable after?

Reply:
Good question. I pushed a patch to put the data back in the input queue, since this would allow an application to handle the error by calling `read_some` / `read_exactly` and then log the faulty data which could be handy to understand why it's malformed. For example if this was used in a web server and someone uploads a raw .png but forgets the request line and headers, the logs could contain enough to quickly spot the problem.

Reply:
That change lgtm